editoast/src/modelsv2/database/connection_pool.rs

```diff
@@ -0,0 +1,275 @@
+    /// // not async, this is equivalent to an infinite loop.
+    /// let conn2 = pool.get_ok();
+    /// ```
+    ///
+    /// 2. If several futures are spawned and each use their own connection, you should make sure
+    ///    that the connection usage follows its acquisition. Failing to do so is equivalent to
+    ///    the following example:
+    ///
+    /// ```rust
+    /// let conn_futures = (0..10).map(|_| async { pool.get() });
+    /// let deadlock = futures::future::join_all(conn_futures).await;
+    /// ```
+    ///
+    /// ## Guidelines
+    ///
+    /// To prevent these issues, prefer the following patterns:
+    ///
+    /// - Don't declare a variable for a single-use connection:
+    ///
+    /// ```rust
+    /// // do
+    /// my_function_using_conn(pool.get().await?.deref_mut()).await;
+    /// // instead of
+    /// let conn = &mut pool.get().await?;
+    /// my_function_using_conn(conn).await;
+    /// ```
+    ///
+    /// - If a connection is used repeatedly, prefer using explicit scoping:
+    ///
+    /// ```rust
+    /// let my_results = {
+    ///     let conn = &mut pool.get().await?;
+    ///     foo(conn).await + bar(conn).await
+    /// };
+    /// // you may acquire a new connection afterwards
+    /// ```
+    ///
+    /// - If you need to open several connections, then the connection must be
+    ///   acquired just before its usage, and dropped just after, **all in the same future**.
+    ///   And these fututres must all be awaited before attempting to acquire a new connection.
+    ///
+    /// ```rust
+    /// let operations =
+    ///     items.into_iter()
+    ///         .zip(pool.iter_conn())
+    ///         .map(|(item, conn)| async {
+    ///             let conn = conn.await?; // note the await here
+    ///             item.do_something(conn).await
+    ///         });
+    /// let results = futures::future::try_join_all(operations).await?;
+    /// // you may acquire a new connection afterwards
+    /// ```
+    pub async fn get(&self) -> Result<DbConnectionV2, DbConnectionError> {
+        #[cfg(test)]
+        let conn = self.get_test().await;
+        #[cfg(not(test))]
+        let conn = self.get_proxy().await;
+        conn
+    }
+
+    /// Returns an infinite iterator of futures resolving to connections acquired from the pool
+    ///
+    /// Meant to be used in conjunction with `zip` in order instanciate a bunch of tasks to spawn.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// let operations =
+    ///     items.into_iter()
+    ///         .zip(pool.iter_conn())
+    ///         .map(|(item, conn)| async {
+    ///             let conn = conn.await?; // note the await here
+    ///             item.do_something(conn).await
+    ///         });
+    /// let results = futures::future::try_join_all(operations).await?;
+    /// // you may acquire a new connection afterwards
+    /// ```
+    #[allow(unused)] // TEMPORARY
```

Comment:
Maybe say why it's temporary so if we ever forget to remove it, the future developer who's discovering this know if it can be removed or not.

```suggestion
    #[allow(unused)] // TEMPORARY until ModelV2 is used
```